prog/prog.go

```diff
@@ -166,8 +169,9 @@ func (target *Target) PhysicalAddr(arg *PointerArg) uint64 {
 // Used for BufferType.
 type DataArg struct {
 	ArgCommon
-	data []byte // for in/inout args
-	size uint64 // for out Args
+	compressed bool
```

Comment:
I think this belongs to the arg type.
See how generation/mutation/encoding of DataArg depend on the type:
https://github.com/google/syzkaller/blob/master/prog/mutation.go#L352-L379
https://github.com/google/syzkaller/blob/master/prog/encoding.go#L160
We could add BufferFSImage type and add  isCompressedDataType function (or perhaps better to make it methods on the BufferType.